### PR TITLE
Add minor fixes for code formatting and comments when using the snip plugin

### PR DESCRIPTION
**Pull request by bdemers**

### CODE REVIEW

File: code-snippet-plugin/src/test/java/com/okta/maven/snippet/Example.java [hunk 1]
```diff
@@ -18,12 +18,17 @@
 public class Example {
 
     public void doFoo() {
-        System.out.println("doing foo");
+        System
+          .out.println("doing foo");
```
Comment (jmelberg-okta) [inline code-snippet-plugin/src/test/java/com/okta/maven/snippet/Example.java]: Was this change intended? Seems that this `System.out.println` can fit any width-rules.

File: code-snippet-plugin/src/test/java/com/okta/maven/snippet/Example.java [hunk 2]
```diff
@@ -18,12 +18,17 @@
...
         int result = 1 + 1;
     }
 
     private void andBar(String value) {
+        // with a comment
         String another = value + "bar";
+
+        // another comment
         int result = 2 + 2;
+
```
Comment (jmelberg-okta) [inline code-snippet-plugin/src/test/java/com/okta/maven/snippet/Example.java]: nit: Extra line

File: code-snippet-plugin/src/main/java/com/okta/maven/snippet/CodeSnippetMojo.java [hunk 3]
```diff
@@ -115,15 +118,30 @@ public void execute() throws MojoExecutionException {
...
                 super.visit(method, arg);
 
                 String body = method.getBody().get().getStatements().stream()
-                        .map(Node::toString)
+                        .map(statement -> {
+                            TokenRange tokenRange = statement.getTokenRange().get();
+                            Optional<Comment> comment = statement.getComment();
+
+                            StringBuilder blockAString = new StringBuilder();
+                            if ( comment.isPresent()) {
```
Comment (jmelberg-okta) [inline code-snippet-plugin/src/main/java/com/okta/maven/snippet/CodeSnippetMojo.java]: `if ( comment.isPresent())` -> `if (comment.isPresent())`
^ Remove the extra space

File: code-snippet-plugin/src/main/java/com/okta/maven/snippet/CodeSnippetMojo.java [hunk 4]
```diff
@@ -115,15 +118,30 @@ public void execute() throws MojoExecutionException {
...
             public void visit(final MethodDeclaration method, final Object arg) {
                 super.visit(method, arg);
 
                 String body = method.getBody().get().getStatements().stream()
-                        .map(Node::toString)
+                        .map(statement -> {
+                            TokenRange tokenRange = statement.getTokenRange().get();
+                            Optional<Comment> comment = statement.getComment();
+
+                            StringBuilder blockAString = new StringBuilder();
```
Comment (jmelberg-okta) [inline code-snippet-plugin/src/main/java/com/okta/maven/snippet/CodeSnippetMojo.java]: Are you expecting to have multiple block strings in this method? If not, maybe we can change `blockAString` to `blockString` to make this more clear.

File: code-snippet-plugin/src/main/java/com/okta/maven/snippet/CodeSnippetMojo.java [hunk 5]
```diff
@@ -115,15 +118,30 @@ public void execute() throws MojoExecutionException {
...
-
         new VoidVisitorAdapter<Object>() {
             @Override
             public void visit(final MethodDeclaration method, final Object arg) {
                 super.visit(method, arg);
 
                 String body = method.getBody().get().getStatements().stream()
-                        .map(Node::toString)
+                        .map(statement -> {
+                            TokenRange tokenRange = statement.getTokenRange().get();
```
Comment (jmelberg-okta) [inline code-snippet-plugin/src/main/java/com/okta/maven/snippet/CodeSnippetMojo.java]: Just out of curiosity, why does this method require an extra `get`?

File: code-snippet-plugin/src/main/java/com/okta/maven/snippet/CodeSnippetMojo.java [hunk 6]
```diff
@@ -115,15 +118,30 @@ public void execute() throws MojoExecutionException {
...
+                            StringBuilder blockAString = new StringBuilder();
+                            if ( comment.isPresent()) {
+                                blockAString.append("\n")
+                                            .append(comment.get());
+                            }
+                            blockAString.append(tokenRange.toString().replaceAll(" {8}", ""));
+
+                            return blockAString;
+                        })
                         .collect(Collectors.joining("\n"));
```
Comment (jmelberg-okta) [inline code-snippet-plugin/src/main/java/com/okta/maven/snippet/CodeSnippetMojo.java]: Here, it appears we are joining together this String based on newlines. If the `comment` contains a block string (with newlines needed) - won't this malform our expected output?

As an alternative, could you use a special character sequence `<-->`, or something of the sort as your delimiter?

File: code-snippet-plugin/src/test/java/com/okta/maven/snippet/Example.java (hunk 1 above)
Comment (bdemers) [reply code-snippet-plugin/src/test/java/com/okta/maven/snippet/Example.java]: This is the input for a test, to make sure the output ends up in the same format

File: code-snippet-plugin/src/test/java/com/okta/maven/snippet/Example.java (hunk 2 above)
Comment (bdemers) [reply code-snippet-plugin/src/test/java/com/okta/maven/snippet/Example.java]: same as above

File: code-snippet-plugin/src/main/java/com/okta/maven/snippet/CodeSnippetMojo.java (hunk 4 above)
Comment (bdemers) [reply code-snippet-plugin/src/main/java/com/okta/maven/snippet/CodeSnippetMojo.java]: ha, that was suppose to be `blockAsString` good catch :)

File: code-snippet-plugin/src/main/java/com/okta/maven/snippet/CodeSnippetMojo.java (hunk 5 above)
Comment (bdemers) [reply code-snippet-plugin/src/main/java/com/okta/maven/snippet/CodeSnippetMojo.java]: It is a Java Optional. Something like:
```java
Optional<String> myValue = ...
String actualValue = myValue.get(); // gets the value or throws an exception
```
It is a functional technique to not null check all over the place. In my case I'm not handling the error because i want the build to fail if there isn't anything to `get`.

File: code-snippet-plugin/src/main/java/com/okta/maven/snippet/CodeSnippetMojo.java (hunk 6 above)
Comment (bdemers) [reply code-snippet-plugin/src/main/java/com/okta/maven/snippet/CodeSnippetMojo.java]: This joins multiple statements together. given the two statements:
```
int one = 1;
```
and
```
int two = 2;
```
The output would match what is in the source file:

```
int one = 1;
int two = 2;
```

File: code-snippet-plugin/src/main/java/com/okta/maven/snippet/CodeSnippetMojo.java (hunk 4 above)
Comment (bdemers) [reply code-snippet-plugin/src/main/java/com/okta/maven/snippet/CodeSnippetMojo.java]: Note the statement could be a single line or a multiple lines:

Single: `int foo = 1;`

or  multiple:

```java
 System
          .out.println("doing foo");
```

The second looks funny, but it is for testing purposes, a real example would look more like:
```java
SwaApplication swaApp = client.instantiate(SwaApplication.class)
    .setSettings(client.instantiate(SwaApplicationSettings.class)
    .setApp(client.instantiate(SwaApplicationSettingsApplication.class)
        .setButtonField("btn-login")
        .setPasswordField("txtbox-password")
        .setUsernameField("txtbox-username")
        .setUrl("https://example.com/login.html")));
```

This is a single statement.

It would be nicer to just get the raw text from the method block, but this was a relatively quick solution to the problem (given the API of the JavaParser)

**Comment by bdemers** [comment]
@jmelberg-okta I should have included this earlier, but this is a more concrete usage:
Where this source file: https://github.com/okta/okta-sdk-java/blob/41d147e0acc488385dc155c404121fa8e49be593/examples/quickstart/src/main/java/quickstart/ReadmeSnippets.java

Is rendered into the README:

https://github.com/okta/okta-sdk-java/blob/41d147e0acc488385dc155c404121fa8e49be593/README.md#usage-guide
